### PR TITLE
added support for all type of responses

### PR DESCRIPTION
**Pull request by AGalabov**

### CODE REVIEW

File: src/openapi-generator.ts [hunk 1]
```diff
@@ -86,6 +95,23 @@ export class OpenAPIGenerator {
     };
   }
 
+  private sortDefinitions() {
```
Comment (georgyangelov) [inline src/openapi-generator.ts:98]: I see now. This should be ok, my other idea was to do a pass like this:

```ts
this._definitions.forEach((def) => {
  switch (def.type) {
    case 'schema': this.schemaRefs[def.refId] = def; break;
    case 'parameter': this.paramRefs[def.refId] = def; break;
    case 'route': this.routeRefs[def.refId] = def; break;
  }
});
```

**Review by georgyangelov — APPROVED** [review]

File: src/openapi-generator.ts [hunk 2]
```diff
@@ -382,6 +402,42 @@ export class OpenAPIGenerator {
...
+      return {
+        description: metadata.description,
+      };
+    }
+
+    const metadata = this.getMetadata(response.schema);
+    const responseSchema = this.generateInnerSchema(response.schema);
+
+    if (!metadata?.description) {
+      throw new Error(
```
[the remark this answers is not in the document]
Comment (AGalabov) [reply src/openapi-generator.ts:426]: Yes it is. Responses need the description key. Typescript would also complain - `ResponseObject` requires it correctly as well

File: src/openapi-generator.ts [hunk 3]
```diff
@@ -151,42 +170,52 @@ export class OpenAPIGenerator {
       const parameters = Object.entries(propTypes).map(([key, schema]) => {
         const innerMetadata = this.getMetadata(schema);
 
-        if (innerMetadata?.name && innerMetadata.name !== key) {
+        const innerParameterMetadata = innerMetadata?.param;
```
Comment (georgyangelov) [inline src/openapi-generator.ts:173]: We should document this somewhere in the README

File: src/openapi-registry.ts [hunk 4]
```diff
@@ -18,8 +18,9 @@ export interface RouteConfig extends Partial<OperationObject> {
     query?: ZodObject<any>;
     headers?: ZodType<unknown>[];
   };
-  response: ZodType<unknown>;
-  errors?: any[];
+  responses: {
+    [key: number]: ResponseConfig;
```
Comment (georgyangelov) [inline src/openapi-registry.ts:22]: Wasn't there a `default` kеy somewhere? Also JS object keys can only be strings, so this is a bit weird

File: src/openapi-generator.ts (hunk 3 above)
Comment (AGalabov) [reply src/openapi-generator.ts:173]: Yup, I'll add it in the Readme.md. I'll probably separate them into:
1. Defining schemas
2. Defining parameters
3. Defining routes

File: src/openapi-registry.ts (hunk 4 above)
Comment (AGalabov) [reply src/openapi-registry.ts:22]: Hmm there is a default key, yes. I guess it should be closer to the type `ResponsesObject`
```
    default?: ResponseConfig;
    [statuscode: string]: ResponseConfig;
```

Comment (AGalabov) [reply src/openapi-registry.ts:22]: And that's basically: `[status: string]: ResponseConfig;`

**Pull request merged**